helpers.js

```diff
@@ -126,11 +129,14 @@ function refLinks(options) {
     refMap.set(name, slug);
   }
 
+  function formatLink(match, brackets, title, parentheses, link) {
+    const ref = safeRef(title, link);
+    return brackets + (ref ? `[${ref}]` : '');
+  }
+
   const content = options.fn(this)
-    .replace(reInlineLinks, (match, brackets, title, parentheses, link) => {
-      const ref = safeRef(title, link);
-      return brackets + (ref ? `[${ref}]` : '');
-    });
+    .replace(reInlineLinks, formatLink)
+    .replace(reInlineSigLinks, formatLink);
```

Comment:
do you want to do multiple replaces like this or just have a `|` in your regex? This way will cascade replaces between them (i.e. the output of the first replace is used to do the second), whereas a pipe in the regex will just do one big replace but with double the function arguments :(

Reply:
I was unsuccessful with using a pipe in my regex to handle all scenarios being tested... I attempted that first but resorted to this approach.

Reply:
Oh, I see what you're saying. One regex with both full expressions. Yea, dunno if that would be any faster, certainly messier with the fn args. I didn't experience any noticely slowdown with the douple replace here.

Reply:
Oh, I see what you're saying. One regex with both full expressions. Yea, dunno if that would be any faster, certainly messier with the fn args. I didn't experience any noticely slowdown with the douple replace here.

Reply:
the issue I see would be something like this:

```js
const regex1 = /foo/g;
const regex2 = /bar/g;
const combined = /foo|bar/g;
console.log('fooar'.replace(regex1, 'foob').replace(regex2, 'foob')); // -> 'foofoob'
console.log('fooar'.replace(combined, 'foob'); // -> 'foobar'
```

Not sure if something like that can happen in this instance though, my brain isn't able to parse those regexes today. 🤣

Reply:
I'll add a test to match sure that situation works.

I'll noodle with the regex some more tonight to see if I can consolidate it to one expression.

Reply:
So @msluther I took another stab at a single regex again without success. Basically, we have 2 categories of link titles: normal and signature. The normal links can have multiple per line, and have text with escaped brackets in line. The first regex handles this. Signatures are the special case which can be assumed as 1 per line, and need to allow for `([])` syntaxes in the title.

So, as the tests show now, my current approach is safe, so to speak, given the known generated output. Should this output change, I think the correct solution will be to implement or try to integrate a proper markdown link parser.

I don't find that likely, so I think these two regular expressions for the expected categories of generated links should be fine.